server/main_compute.py

```
@@ -266,6 +266,15 @@ def verify_internal_api_secret():
 # Separate from _OPEN_PREFIXES: this only skips routes that carry identity headers before auth completes (login/register); webhook routes don't need listing here because they lack X-User-ID/X-Org-ID and the check below short-circuits on missing headers.
 _TENANT_OPEN_PREFIXES = ("/api/auth/login", "/api/auth/register", _HEALTH_PATH)
 
+
+def _audit_tenant_failure(user_id, org_id, action, reason):
```

Ruff (0.15.11)

[warning] Missing return type annotation for private function `_audit_tenant_failure`

Add return type annotation: `None`

(ANN202)